### ISSUE TITLE
Ability to disable table caption in FlexmarkHtmlParser

### ISSUE THREAD

**Issue by dmitrymurashenkov** [vsch/flexmark-java#318]
**Is your feature request related to a problem? Please describe.** 

I'm converting html to markdown to store it and then render as html again. The toolset used for rendering is not flexmark-based and doesn't support caption for markdown tables. Caption gets rendered as [Caption text here] which might not be desirable.

**Describe the solution you'd like**

It would be nice to have an option to disable table captions altogether or configure how they are translated to markdown.

Seems that this issue should be implemented after #313 is closed and some generic way to customize conversion process is provided.

**Comment by vsch** [comment]
@dmitrymurashenkov, I will add it as a bug. It should use the TableExtensions options to determine what type of table elements to generate. TableExtension has an option for captions.

That way you setup your parser extension options of what you want to parse and use them for the HTML to Markdown conversion. It should generate Markdown that the given Parser configuration can parse and generate HTML representative of the original HTML within Markdown limitations.

**Comment by vsch** [comment]
@dmitrymurashenkov, actually the code is already there. All table options from formatter can be used since the html parser uses the formatter to output the final table.

The html parser table options are just convenience alias for the formatter options:

```java
    public static final DataKey<Integer> TABLE_MIN_SEPARATOR_COLUMN_WIDTH = TableFormatOptions.FORMAT_TABLE_MIN_SEPARATOR_COLUMN_WIDTH;
    public static final DataKey<Integer> TABLE_MIN_SEPARATOR_DASHES = TableFormatOptions.FORMAT_TABLE_MIN_SEPARATOR_DASHES;
    public static final DataKey<Boolean> TABLE_LEAD_TRAIL_PIPES = TableFormatOptions.FORMAT_TABLE_LEAD_TRAIL_PIPES;
    public static final DataKey<Boolean> TABLE_SPACE_AROUND_PIPES = TableFormatOptions.FORMAT_TABLE_SPACE_AROUND_PIPES;
    public static final DataKey<TableCaptionHandling> TABLE_CAPTION = TableFormatOptions.FORMAT_TABLE_CAPTION;
    public static final DataKey<Boolean> LISTS_END_ON_DOUBLE_BLANK = new DataKey<>("LISTS_END_ON_DOUBLE_BLANK", false);
```

I added the `TABLE_CAPTION` to the HTML parser options for the next release but you can just as well use `TableFormatOptions.FORMAT_TABLE_CAPTION` option in the current version and set it to: `TableCaptionHandling.REMOVE`

It should render the table without the caption.

**Comment by vsch** [comment]
Fix for this is available. Repo updated, maven updated but may take a while to show up in maven central.
